pkg/cmd/deploy/upload.go

```
@@ -1,136 +1,282 @@
+		return fmt.Errorf("failed to open file %s: %w", filePath, err)
 	}
+	defer file.Close()
 
-	if err := cmd.FilepathWalk(pathStatic, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
+	fileName := filepath.Base(filePath)
+
+	uploadInput := &s3.PutObjectInput{
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(fileName), // Name of the file in the bucket
+		Body:   file,
+	}
+
+	_, err = s3Client.PutObject(ctx, uploadInput)
+	if err != nil {
+		return fmt.Errorf("failed to upload file to bucket %s: %w", bucketName, err)
+	}
+
+	return nil
+}
+
+// Worker responsible for zipping and uploading files
+func worker(ctx context.Context, cfg aws.Config, filesChan <-chan string,
+	wg *sync.WaitGroup, baseDir string, workerID int, atomicCounter *int32,
+	bucketName string, progressBar *progressbar.ProgressBar) {
+	defer wg.Done()
+
+	var (
+		currentSize int64
+		filesToZip  []string
+	)
+
+	zipFileName := fmt.Sprintf("project_part_%d.zip", workerID)
 
-		// Skip node_modules and .edge directories
-		if info.IsDir() && (strings.Contains(path, "node_modules") || strings.Contains(path, ".edge")) {
-			logger.Debug("Skipping directory: " + path)
-			return filepath.SkipDir
+	for file := range filesChan {
+		filesToZip = append(filesToZip, file)
+
+		// Get the file size and add it to the current size
+		fileInfo, err := os.Stat(file)
+		if err != nil {
+			log.Printf("Failed to get file info for %s: %v", file, err)
+			continue
 		}
+		currentSize += fileInfo.Size()
 
-		if !info.IsDir() {
-			fileContent, err := cmd.Open(path)
+		// If the current size exceeds the limit (1MB), zip and upload
+		if currentSize >= maxZipSize {
+			err = zipAndUpload(ctx, cfg, filesToZip, zipFileName, baseDir,
+				atomicCounter, bucketName, progressBar)
 			if err != nil {
-				logger.Debug("Error while trying to read file <"+path+"> about to be uploaded", zap.Error(err))
-				return err
+				log.Printf("Failed to zip and upload files: %v", err)
```

Change this to a variable and return the error variable
Also: you SHOULD return the error, to stop the execution :)